### PR TITLE
fix: Separately query server for featured events

### LINKED ISSUE #4481: Featured events no longer on start page
A number of upcoming events are no longer listed on the start page in the section Upcoming or Featured Events.

Screenshot from 2020-06-21 06-14-09

See featured events at https://eventyay.com/admin/events/live

Screenshot from 2020-06-21 06-13-41

### PR DESCRIPTION
**Pull request by Haider8**
Fixes #4481

#### Short description of what this resolves:
Some featured events was no longer on start page because they are being queried on the frontend from already existing upcoming events which puts the constraint on the event's date.

#### Changes proposed in this pull request:
Separately query server for featured events by using `is-featured` filter.

featured-events

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/routes/index.js [hunk 1]
```diff
@@ -132,13 +132,40 @@ export default class IndexRoute extends Route {
...
+                val  : 'public'
+              },
+              {
+                name : 'is-featured',
+                op   : 'eq',
+                val  : true
+              }
+            ]
+          }
+        ]
```
Comment (iamareebjamal) [inline app/routes/index.js:160]: This shows all featured events including the ones from past

Comment (Haider8) [reply app/routes/index.js:170]: Done

Comment (iamareebjamal) [reply app/routes/index.js:170]: Nope. Reuse the returned filterOptions. This will break on search page

**Comment by iamareebjamal** [comment]
Since call for speaker filtering using server will be difficult, change the computed property to filter those from combined featured and upcoming events

File: app/controllers/index.js [hunk 2]
```diff
@@ -14,17 +14,14 @@ export default class IndexController extends Controller {
 
   @computed('filteredEvents.[]')
```
Comment (iamareebjamal) [inline app/controllers/index.js:15]: You haven't listed featured events in computed dependency

File: app/controllers/index.js [hunk 3]
```diff
@@ -14,17 +14,14 @@ export default class IndexController extends Controller {
...
-      const callForPapers = event.get('speakersCall');
-      const sessionEnabled = event.isSessionsSpeakersEnabled;
-      if (!callForPapers || !callForPapers.get('startsAt')  || !callForPapers.get('endsAt')) {
-        return false;
-      }
-      const startDateTime = callForPapers.get('startsAt');
-      const endDateTime = callForPapers.get('endsAt');
-      const privacyState = callForPapers.get('privacy');
-      return (moment().isBetween(startDateTime, endDateTime) && (sessionEnabled) && (privacyState === 'public'));
+    let filteredEventsCfs = this.filteredEvents.filter(event => {
```
Comment (iamareebjamal) [inline app/controllers/index.js:17]: Why let and not const?

Comment (iamareebjamal) [inline app/controllers/index.js:17]: You can directly pass `isEventCfsOpen` in filter function, no need to create an arrow function

Comment (Haider8) [reply app/controllers/index.js]: Done

**Pull request merged**